### PR TITLE
update regexp compability

### PR DESCRIPTION
**Pull request by xzhangxian1008**
### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v6.3 (TiDB 6.3 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: functions-and-operators/string-functions.md [hunk 1]
```diff
@@ -95,3 +95,11 @@ TiDB 与 MySQL 在 `match_type` 上的差异：
     例如：MySQL 中 `SELECT REGEXP_LIKE(a, b, "n") FROM t1;` 在 TiDB 中需要修改为 `SELECT REGEXP_LIKE(a, b, "s") FROM t1;`。
 
 - TiDB 不支持 `match_type` 为 `"u"`。
+
+### 数据类型兼容性
+
+[MySQL文档](https://dev.mysql.com/doc/refman/8.0/en/regexp.html#function_regexp-like:~:text=use%20of%20a%20binary%20string%20with%20any%20of%20the%20MySQL%20regular%20expression%20functions%20is%20rejected)显示正则函数不会支持binary string，但在实际使用过程中，假如所有参数或者返回值的数据类型都是binary string，正则函数是可以正常使用的，否则报错。TiDB目前完全禁止使用binary string，无论什么情况都会报错。
```
Comment (Oreoxmt) [inline functions-and-operators/string-functions.md:101]:
```suggestion
TiDB 与 MySQL 在二进制字符串 (binary string)数据类型上的差异：

- MySQL 8.0.22 及以上版本中正则函数不支持二进制字符串，具体信息可查看 [MySQL 文档](https://dev.mysql.com/doc/refman/8.0/en/regexp.html)。但在实际使用过程中，如果所有参数或者返回值的数据类型都是二进制字符串，则正则函数可以正常使用，否则报错。
- TiDB 目前完全禁止使用二进制字符串，无论什么情况都会报错。
```

File: functions-and-operators/string-functions.md [hunk 2]
```diff
@@ -95,3 +95,11 @@ TiDB 与 MySQL 在 `match_type` 上的差异：
...
+
+### 其它兼容性
+
+- 对于`regexp_replace("", "^$", "123")`这样替换空字符串的操作，MySQL是不会对空串进行替换的，但是TiDB会选择替换，所以在本例中，MySQL和TiDB的结果分别为`""`和`"123"`
```
Comment (Oreoxmt) [inline functions-and-operators/string-functions.md:105]:
```suggestion
TiDB 与 MySQL 在替换空字符串上的差异：

下面以 `REGEXP_REPLACE("", "^$", "123")` 为例：

- MySQL 不会对空串进行替换，其结果为 `""`。
- TiDB 会对空串进行替换，其结果为 `"123"`。
```

**Review by windtalker — APPROVED** [review]
LGTM

File: functions-and-operators/string-functions.md [hunk 3]
```diff
@@ -95,3 +95,19 @@ TiDB 与 MySQL 在 `match_type` 上的差异：
     例如：MySQL 中 `SELECT REGEXP_LIKE(a, b, "n") FROM t1;` 在 TiDB 中需要修改为 `SELECT REGEXP_LIKE(a, b, "s") FROM t1;`。
 
 - TiDB 不支持 `match_type` 为 `"u"`。
+
+### 数据类型兼容性
+
+TiDB 与 MySQL 在二进制字符串 (binary string)数据类型上的差异：
```
Comment (Oreoxmt) [inline functions-and-operators/string-functions.md:101]:
```suggestion
TiDB 与 MySQL 在二进制字符串 (binary string) 数据类型上的差异：
```

**Review by Oreoxmt — APPROVED** [review]
Rest LGTM

**Review by shichun-0415 — APPROVED** [review]
LGTM

**Pull request merged**